### PR TITLE
Investo 130 Order Add Native

### PR DESCRIPTION
**Pull request by yhoisa**

### CODE REVIEW

**Review by AlekseyP18 — APPROVED** [review]

File: src/collections/order/order.collection.native.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  getUserOrderCollection<T>(userId).doc(id);
+
+export const addOrder = async (data: OrderInputInterface) => {
+  const batch = getStore<CommonNativeInterface>().batch();
+
+  const orderNewReference = getOrderCollection<OrderInputInterface>().doc();
+  const userOrderReference = getUserOrderReference(data.userId, orderNewReference.id);
+
+  const order = {
+    createdAt: firestore.FieldValue.serverTimestamp(),
```
Comment (ihorHaiduk) [inline src/collections/order/order.collection.native.ts:25]: baseData

File: src/collections/order/order.collection.native.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export const getUserOrderCollection = <T = OrderDocumentInterface>(userId: string) =>
+  getStore<CommonNativeInterface>().collection<T>(CollectionsEnum.Users).doc(userId).collection(CollectionsEnum.Orders);
+export const getUserOrderReference = <T = OrderDocumentInterface>(userId: string, id: string) =>
+  getUserOrderCollection<T>(userId).doc(id);
+
+export const addOrder = async (data: OrderInputInterface) => {
+  const batch = getStore<CommonNativeInterface>().batch();
+
+  const orderNewReference = getOrderCollection<OrderInputInterface>().doc();
```
Comment (ihorHaiduk) [inline src/collections/order/order.collection.native.ts:21]: getOrderReference()

File: src/collections/order/order.collection.native.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export const cancelOrder = (id: string) =>
+  getStore<CommonNativeInterface>().runTransaction(async transaction => {
+    const orderDoc = await transaction.get(getOrderReference(id));
+
+    if (!orderDoc.exists) throw new Error('Order does not exist!');
+
+    const order = orderMap(orderDoc);
+    const orderUpdate = {
+      canceledAt: firestore.FieldValue.serverTimestamp(),
```
Comment (ihorHaiduk) [inline src/collections/order/order.collection.native.ts:47]: baseDataCanceled

File: src/hooks/order/use-order-add.hook.native.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { useBehavior } from '../utils/behaviour.hook';
+import { useError } from '../utils/error.hook';
+import { useObservable } from '../utils/observable.hook';
+import { useRequest } from '../utils/request.hook';
+
+type OrderAddHandlerType = (form: OrderInputInterface | null) => void;
+type RequestAddType = ReturnPromiseType<typeof addOrder>;
+
+export const useOrderAdd = (): [OrderAddHandlerType, string, boolean, Error | null] => {
+  const [form$, handleCreateOrder] = useBehavior<OrderInputInterface | null>(null);
```
Comment (ihorHaiduk) [inline src/hooks/order/use-order-add.hook.native.ts:17]: (null, false)

File: src/hooks/order/use-order-add.hook.native.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  const [errorValidation, handleError] = useError<OrderInputInterface>();
+  const [handleAddRequest, isLoadingAdd, errorAdd] = useRequest<RequestAddType>();
```
Comment (ihorHaiduk) [inline src/hooks/order/use-order-add.hook.native.ts:19]: handleRequest, isLoading, error

**Pull request closed without merging**